x-pack/plugins/features/common/feature.ts

```diff
@@ -122,8 +122,11 @@ export interface FeatureConfig {
    * @private
    */
   reserved?: {
-    privilege: FeatureKibanaPrivileges;
     description: string;
+    privileges: Array<{
+      id: string;
```

Comment:
Since the new privilege will have an ID of `reserved_${id}`, what do you think about changing this property to `idSuffix` to make it more descriptive?

Reply:
> Since the new (serialized) privilege will have an ID of `reserved_${id}`, what do you think about changing this property to `idSuffix` to make it more descriptive?

I'm tempted to leave it as-is, because that'll be consistent with the sub-feature privileges that we register today. Features generally aren't aware of their serialized privilege format.

e.g. Feature `foo` has sub-feature privilege `bar`. The serialized privilege becomes `feature_foo.bar`

Reply:
> Features generally aren't aware of their serialized privilege format.

OK, I wasn't sure about that. I agree it makes sense to leave it as-is, then!

Reply:
- not sure if better, but did you think about using a Record<id, privilege> instead of an Array<{id, privilege}>? (even if the array seems more futur-proof)
- if sticking to an an array, I would probably introduce a type for this structure.

Reply:
> * not sure if better, but did you think about using a Record<id, privilege> instead of an Array<{id, privilege}>? (even if the array seems more futur-proof)
> * if sticking to an an array, I would probably introduce a type for this structure.

I didn't explore this route, but I don't mind doing so if you'd prefer this approach. Otherwise I'm happy to introduce an explicit type for this structure.

I'm not too concerned with future-proofing this, because we want to eliminate the concept of reserved privileges once ML and Monitoring are able to migrate off of them.

Reply:
Imho current implem is fine, don't see any major reason to change now. Just adding a proper type SGTM.